Ps3DiscDumper/Dumper.cs

```diff
@@ -151,27 +151,34 @@ private void CheckParamSfo(ParamSfo sfo)
             Log.Info($"Game title: {Title}");
         }
 
-        public void DetectDisc(Func<Dumper, string> outputDirFormatter = null)
+        public void DetectDisc(string inDir = "", Func<Dumper, string> outputDirFormatter = null)
         {
             outputDirFormatter = outputDirFormatter ?? (d => $"[{d.ProductCode}] {d.Title}");
             string discSfbPath = null;
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                var drives = DriveInfo.GetDrives().Where(d => d.DriveType == DriveType.CDRom && d.IsReady);
-                foreach (var drive in drives)
+                if (!String.IsNullOrEmpty(inDir))
+                    discSfbPath = Path.Combine(inDir, "PS3_DISC.SFB");
```

Comment:
you leave `Drive` uninitialized this way, which will fail later when you will try to actually dump the disc

for windows effectively all you can pass is the drive letter (e.g. `F:`), so it makes sense to check if you have such drive letter after enumerating all of them, instead of doing if-else outside it